arcus/exc.py

```diff
@@ -35,3 +35,8 @@ def __repr__(self):
         return self.__class__.__name__ + ', '.join(
             [f'{attr}={repr(value)}'
              for attr, value in self.__dict__.items()]) + ')'
+
+
+class InvalidOperation(UnprocessableEntity):
+    def __init__(self, txn_id: Union[int, str]):
+        self.message = f'Unable to cancel the transaction {txn_id}'
```

Comment:
this should call `super().__init__(code, transaction_id=transaction_id)`.

please don't use abbreviations like txn for transaction in function / method params. It's clearer to be explicit.

for adding the message, take a look at the botmaker client code for how it adds message to the parent exception and prints it if available.